src/ray/core_worker/core_worker.cc

```diff
@@ -459,6 +467,17 @@ Status CoreWorker::SubmitTask(const RayFunction &function,
   const auto task_id =
       TaskID::ForNormalTask(worker_context_.GetCurrentJobID(),
                             worker_context_.GetCurrentTaskID(), next_task_index);
+
+  if (task_options.is_direct_call) {
+    // TODO(ekl) offload task building onto a thread pool for performance
```

Comment:
Not sure about this. Consider the case when a user allocates one core to a task that they think is single threaded but submits many tasks - they would be unknowingly consuming more cores and degrading performance for other tasks/actors.

Reply:
Well, we don't really have a choice unless we are happy with ~20k at most. Btw, CPU oversubscription doesn't really matter unless you are overloading to by a huge amount.